esi_ui/static/dashboard/project/esi/nodes/esi-nodes-table.controller.js

```diff
@@ -204,21 +205,49 @@
       var launchContext = {
         node: node
       };
-
+    
       manageNetworksModalService.open(launchContext)
-      .then(function (response) {
-        if (response.action === 'attach') {
-          nodesService.networkAttach(node, response.attach)
-          .then(function() {
-            init();
-          });
-        } else {
-          nodesService.networkDetach(node, response.detach)
-          .then(function() {
-            init();
-          });
-        }
-      });
+        .then(function (response) {
+          if (response.action === 'attach') {
+            node.network_operation = 'attaching';
+            toastService.add('success', 'Attaching network, please wait...');
+            
+            nodesService.networkAttach(node, response.attach)
+              .then(function() {
+                init();
```

Comment:
the init() function returns a Promise, so instead of using $timeout, you can use a .then()